ec3730/Controllers/PingViewController.swift

```diff
@@ -209,31 +209,40 @@ struct PingSetView: View {
 @available(iOS 14.0, *)
 struct PingSetList: View {
     @Environment(\.managedObjectContext) private var viewContext
-
+    @State var mode: EditMode = .inactive
     @FetchRequest(fetchRequest: PingSet.fetchAllRequest()) var pings: FetchedResults<PingSet>
 
     var body: some View {
-        List {
-            ForEach(pings) { ping in
-                NavigationLink(
-                    destination: PingSetView(ping: ping),
-                    label: {
-                        VStack(alignment: .leading) {
-                            HStack(alignment: .center) {
-                                Text("\(ping.host)").font(.headline)
-                                Spacer()
-                                Text("\(ping.pings.count)").font(.footnote).foregroundColor(.gray)
+        VStack {
+            if mode == .active, pings.count > 1 {
+                Button {
+                    deleteAllItems()
+                } label: {
+                    Text("Delete All")
+                }
+            }
+            List {
+                ForEach(pings) { ping in
+                    NavigationLink(
+                        destination: PingSetView(ping: ping),
+                        label: {
+                            VStack(alignment: .leading) {
+                                HStack(alignment: .center) {
+                                    Text("\(ping.host)").font(.headline)
+                                    Spacer()
+                                    Text("\(ping.pings.count)").font(.footnote).foregroundColor(.gray)
+                                }
+                                Text("\(ping.timestamp)").font(.caption)
                             }
-                            Text("\(ping.timestamp)").font(.caption)
                         }
-                    }
-                )
-            }.onDelete(perform: deleteItems)
-        }.listStyle(PlainListStyle()).navigationTitle("History").toolbar {
-            #if os(iOS)
-                EditButton()
-            #endif
-        }
+                    )
+                }.onDelete(perform: deleteItems)
+            }.listStyle(PlainListStyle()).navigationTitle("History").toolbar {
+                #if os(iOS)
```

Comment:
Did it complain about this? Why the `#if os(iOS)`?

Reply:
It was already there in code. I didn't add but it does't complain.